### PR TITLE
Setup CI scripts

### PR DESCRIPTION
**Pull request by cpanato**
This PR add some scripts to be run when any contributor open a PR in this Repo

It will verify for the boilerplate + go mod diff + golang-ci lint + tests

Prow Job presubmit PR: https://github.com/kubernetes/test-infra/pull/19565

Fixes: https://github.com/kubernetes-sigs/mdtoc/issues/3

/cc @saschagrunert @justaugustus @tallclair

### CODE REVIEW

File: hack/verify-go-mod.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# go mod tidy
+# git diff --exit-code
```
Comment (saschagrunert) [inline hack/verify-go-mod.sh:22]: What's the reason that this is commented out?

Comment (cpanato) [reply hack/verify-go-mod.sh:22]: no, forgor to remove the comment.

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**